src/helix/client/client_ext.rs

```diff
@@ -43,21 +44,38 @@ impl<'client, C: crate::HttpClient + Sync + 'client> HelixClient<'client, C> {
     }
 
     /// Get multiple [User](helix::users::User)s from user ids.
-    pub async fn get_users_from_ids<T>(
+    ///
+    /// # Examples
+    ///
+    /// ```rust, no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    /// # let client: helix::HelixClient<'static, twitch_api::client::DummyHttpClient> = helix::HelixClient::default();
+    /// # let token = twitch_oauth2::AccessToken::new("validtoken".to_string());
+    /// # let token = twitch_oauth2::UserToken::from_existing(&client, token, None, None).await?;
+    /// use twitch_api::{helix, types};
+    /// use futures::TryStreamExt;
+    ///
+    /// let users: Vec<helix::users::User> = client
+    ///    .get_users_from_ids(&["1234", "4321"][..].into(), &token).try_collect().await?;
+    /// # Ok(()) }
+    /// ```
+    pub fn get_users_from_ids<T>(
         &'client self,
-        ids: impl AsRef<[&types::UserIdRef]> + Send,
-        token: &T,
-    ) -> Result<Option<helix::users::User>, ClientError<C>>
+        ids: &'client types::Collection<'client, types::UserId>,
+        token: &'client T,
+    ) -> impl futures::Stream<Item = Result<helix::users::User, ClientError<C>>> + Send + Unpin + 'client
     where
         T: TwitchToken + Send + Sync + ?Sized,
     {
-        let ids = ids.as_ref();
-        if ids.len() > 100 {
-            return Err(ClientRequestError::Custom("too many IDs, max 100".into()));
-        }
-        self.req_get(helix::users::GetUsersRequest::ids(ids), token)
-            .await
-            .map(|response| response.first())
+        futures::stream::iter(ids.chunks(100).collect::<Vec<_>>())
+            .map(move |c| {
+                let req = helix::users::GetUsersRequest::ids(c);
+                futures::stream::once(self.req_get(req, token)).boxed()
+            })
+            .flatten_unordered(None)
+            .map_ok(|resp| futures::stream::iter(resp.data.into_iter().map(Ok)))
+            .try_flatten_unordered(None)
     }
```

Comment:
I don't like how the examples have to use `[..]`

I also tried to make the helper functions take `impl Into<Collection<'_, T>>` but it didn't work. Still going to try to make it work but I suspect we need some the changes made in 2024 edition for lifetime captures behind impls, or we go the `TwitchResult` route. I suspect the `TwitchResult` route is what is needed for doing minimal allocations also